### PR TITLE
Medicomp surgery can heal organ damage

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

Yautja do not currently have any ways to heal their organs

# Explain why it's good for the game

This lets them actually fix their organs

# Changelog

:cl:
balance: Adds a way to heal organ damage via the medicomp
/:cl:

### CODE REVIEW

**Comment by Nanu308** [comment]
Actually on second thought, this should rather be added to the yautja self surgery tool, but i'd be fine with Thwei potentially working as Peri, depending on how long Thwei remains in the system

**Comment by Diegoflores31** [comment]
> Actually on second thought, this should rather be added to the yautja self surgery tool, but i'd be fine with Thwei potentially working as Peri, depending on how long Thwei remains in the system

pretty sure a shot of thwei can last like half an hour . so i think surgery is a better alternative

File: code/modules/surgery/mcomp_tendwounds.dm [hunk 1]
```diff
@@ -174,3 +175,114 @@
...
+		target.emote("pain")
+
+	if(user == target)
+		user.visible_message(SPAN_NOTICE("[user] finishes treating the damaged organs on their body with \the [tool]."),
+			SPAN_HELPFUL("You finish treating the damaged organs on your body with \the [tool]."))
+	else
+		user.affected_message(target,
+			SPAN_HELPFUL("You finish treating [target]'s damaged organs with \the [tool]."),
+			SPAN_HELPFUL("[user] finished treating your damaged organs with \the [tool]."),
+			SPAN_NOTICE("[user] finished treating [target]'s damaged organs with \the [tool]."))
```
Comment (harryob) [inline code/modules/surgery/mcomp_tendwounds.dm:258]:
```suggestion
			SPAN_HELPFUL("You finish treating [target]'s damaged organs with [tool]."),
			SPAN_HELPFUL("[user] finished treating your damaged organs with [tool]."),
			SPAN_NOTICE("[user] finished treating [target]'s damaged organs with [tool]."))
```

File: code/modules/surgery/mcomp_tendwounds.dm [hunk 2]
```diff
@@ -174,3 +175,114 @@
...
+
+	if(!istype(tool, /obj/item/tool/surgery/healing_gun))
+		return
+	var/obj/item/tool/surgery/healing_gun/gun = tool
+	gun.loaded = FALSE
+
+	log_interact(user, target, "[key_name(user)] Treated some of [key_name(target)]'s organ damage with \the [tool].")
```
Comment (harryob) [inline code/modules/surgery/mcomp_tendwounds.dm:265]:
```suggestion
	log_interact(user, target, "[key_name(user)] treated some of [key_name(target)]'s organ damage with [tool].")
```

File: code/modules/surgery/mcomp_tendwounds.dm [hunk 3]
```diff
@@ -174,3 +175,114 @@
...
+
+/datum/surgery_step/mtend_wounds/failure(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	log_interact(user, target, "[key_name(user)] failed to tend [key_name(target)]'s organ damage with \the [tool], possibly ending [surgery].")
```
Comment (harryob) [inline code/modules/surgery/mcomp_tendwounds.dm:268]:
```suggestion
	log_interact(user, target, "[key_name(user)] failed to tend [key_name(target)]'s organ damage with [tool], possibly ending [surgery].")
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Issue design-wise: without damage you cannot start the surgery. Check line 17 of mcomp_tendwounds and add a check for organ damage. Otherwise, seems fine.

**Review by morrowwolf — APPROVED** [review]

**Pull request merged**